### ISSUE TITLE
Cannot add an exception to #[deny(lint group)] with #[warn(specific lint)]

### ISSUE THREAD

**Issue by SimonSapin** [rust-lang/rust#41941]
Test case:

```rust
#![deny(warnings)]
#![warn(illegal_floating_point_literal_pattern)]

fn main() {
    if let 0.0 = 1.0 {}
}
```

Expected result: compiles with a warning. 

Actual result with rustc 1.19.0-nightly (978d2cfee 2017-05-10): hard error:

```rust
error: floating-point literals cannot be used in patterns
 --> a.rs:5:12
  |
5 |     if let 0.0 = 1.0 {}
  |            ^^^
  |
  = note: #[deny(illegal_floating_point_literal_pattern)] implied by #[deny(warnings)]
  = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
  = note: for more information, see issue #41620 <https://github.com/rust-lang/rust/issues/41620>
note: lint level defined here
 --> a.rs:1:9
  |
1 | #![deny(warnings)]
  |         ^^^^^^^^

ERROR:rustc_lint::builtin: span mc spanspam
error: aborting due to previous error
```

Note that `allow` *can* add an exception, but of course it makes the warning not be shown at all:

```rust
#![deny(warnings)]
#![allow(illegal_floating_point_literal_pattern)]

fn main() {
    if let 0.0 = 1.0 {}
}
```

Context: I wanted to use this to unblock compiling Servo without completely hiding the problem under a metaphorical carpet.

**Comment by oli-obk** [comment]
related: https://github.com/rust-lang/rust/issues/34596

**Comment by Enselic** [comment]
Triage: This was discussed by the language team [here ](https://github.com/rust-lang/rust/pull/85298#issuecomment-879239928), and it was [decided](https://github.com/rust-lang/rust/pull/85298#issuecomment-879290431) that the current behavior should be kept, but that the documentation should be updated to clarify the situation.

The documentation was updated in [this](https://github.com/rust-lang/rust/pull/46136) PR. After some iteration over the years, we now have [this](https://doc.rust-lang.org/rustc/lints/groups.html) wording:

> Lint Group: warnings - All lints that are set to issue warnings

That makes the current behavior match the docs, so let's close as fixed.
